pkg/controller/scylladbdatacenter/status.go

```diff
@@ -0,0 +1,138 @@
+package scylladbdatacenter
+
+import (
+	"context"
+	"fmt"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+	appsv1 "k8s.io/api/apps/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (scc *Controller) updateStatus(ctx context.Context, currentSC *scyllav1alpha1.ScyllaDBDatacenter, status *scyllav1alpha1.ScyllaDBDatacenterStatus) error {
+	if apiequality.Semantic.DeepEqual(&currentSC.Status, status) {
+		return nil
+	}
+
+	sdc := currentSC.DeepCopy()
+	sdc.Status = *status
+
+	// Make sure that any "live" updates to the status are always manifested in the aggregated fields.
+	updateAggregatedStatusFields(&sdc.Status)
+
+	klog.V(2).InfoS("Updating status", "ScyllaDBDatacenter", klog.KObj(sdc))
+
+	_, err := scc.scyllaClient.ScyllaDBDatacenters(sdc.Namespace).UpdateStatus(ctx, sdc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	klog.V(2).InfoS("Status updated", "ScyllaDBDatacenter", klog.KObj(sdc))
+
+	return nil
+}
+
+func (scc *Controller) getScyllaVersion(sts *appsv1.StatefulSet) (string, error) {
+	firstMemberName := fmt.Sprintf("%s-0", sts.Name)
+	firstMember, err := scc.podLister.Pods(sts.Namespace).Get(firstMemberName)
+	if err != nil {
+		return "", err
+	}
+
+	controllerRef := metav1.GetControllerOfNoCopy(firstMember)
+	if controllerRef == nil || controllerRef.UID != sts.UID {
+		return "", fmt.Errorf("foreign pod")
+	}
+
+	version, err := naming.ScyllaVersion(firstMember.Spec.Containers)
+	if err != nil {
+		return "", err
+	}
+
+	return version, nil
+}
+
+// calculateRackStatus calculates a status for the rack.
+// sts and old status may be nil.
+func (scc *Controller) calculateRackStatus(sdc *scyllav1alpha1.ScyllaDBDatacenter, sts *appsv1.StatefulSet) *scyllav1alpha1.RackStatus {
```

Comment:
Leaving a note here that rack status conditions are dropped from the new API as it wasn't described in the proposal so it confused me for a bit. @zimnx explained it was on purpose as it can still be concluded from the status. I'm fine with the change personally.
fyi @tnozicka

Reply:
They seem to be deduced when mapping the status from ScyllaDBDatacenter to ScyllaCluster so compatibility wise that should be ok, I am fine not having them on the new resource at this point.